### PR TITLE
Add automatic update checking

### PR DESCRIPTION
**Pull request by ellie**
Check for updates, and show a message if there is one available!

I'm going to be rolling out deletion support over several versions, as it's _complicated_ and changes how sync and history is transferred. This way, we make it more likely that people will follow the updates on time.

My Rust is a teeny bit rustier than I'd like, it's been a little while 😞

### CODE REVIEW

File: atuin-client/src/settings.rs [hunk 1]
```diff
@@ -99,31 +106,68 @@ pub struct Settings {
...
+        }
+
+        let value = fs_err::read_to_string(path);
 
-        if !sync_time_path.exists() {
-            return Ok(Utc.ymd(1970, 1, 1).and_hms(0, 0, 0));
+        match value {
+            Ok(v) => Some(v),
+            Err(_) => None,
         }
```
Comment (conradludgate) [inline atuin-client/src/settings.rs:135]:
```suggestion
        value.ok()
```

**Review by conradludgate — APPROVED** [review]
Looking good 🚀 🦄

**Comment by ellie** [comment]
Any idea why the tests are failing @conradludgate?

It's not liking the api client import but I don't get why 🤔

File: atuin-client/src/settings.rs (hunk 1 above)
Comment (ellie) [reply atuin-client/src/settings.rs:135]: Wow this is much neater 🤣

**Comment by conradludgate** [comment]
> It's not liking the api client import but I don't get why 🤔

Ahh yeah. Since that's a sync related feature, it doesn't work in the tests which specify `--no-default-features`.

I split those features out a while back so we could have minimal client-only installs. It shouldn't be too hard to get it to work here either (`needs_update` will just always return None if the feature is disabled)

**Pull request merged**